code/modules/admin/verbs/map_template_loadverb.dm

```diff
@@ -2,19 +2,24 @@
 	set category = "Debug"
 	set name = "Map template - Place"
 
-	var/map = input(usr, "Choose a Map Template to place at your CURRENT LOCATION","Place Map Template") as null|anything in map_templates
-	if(!map)
-		return
+	var/datum/map_template/template
 
 	var/turf/T = get_turf(mob)
 	if(!T)
 		return
-
-	var/datum/map_template/template
-
+	var/map = input(usr, "Choose a Map Template to place at your CURRENT LOCATION","Place Map Template") as null|anything in map_templates
+	if(!map)
+		return
 	template = map_templates[map]
-	template.load(T)
-	message_admins("<span class='adminnotice'>[key_name_admin(usr)] has placed a map template ([template.name]) at <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[T.x];Y=[T.y];Z=[T.z]'>(JMP)</a></span>")
+
+	var/list/preview = list()
+	for(var/S in template.get_affected_turfs(T,centered = TRUE))
+		preview += image('icons/turf/overlays.dmi',S,"greenOverlay")
+	usr.client.images += preview
+	if(alert(usr,"Confirm location.","Template Confirm","Yes","No") == "Yes")
+		template.load(T, centered = TRUE)
+		message_admins("<span class='adminnotice'>[key_name_admin(usr)] has placed a map template ([template.name]) at <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[T.x];Y=[T.y];Z=[T.z]'>(JMP)</a></span>")
+	usr.client.images -= preview
```

Comment:
Will the images in the preview list get GC'ed properly? Or does the turf keep a reference to the image? I mean, I guess it's not that big of a deal even if this did leak memory, but still.